lib/puppet/provider/firewall/iptables.rb

```diff
@@ -472,7 +474,7 @@ def self.rule_to_hash(line, table, counter)
     # Convert booleans removing the previous cludge we did
     @known_booleans.each do |bool|
       if hash[bool] != nil then
-        if hash[bool] != "true" then
+        if hash[bool] != "true" && hash[bool] != "!" then
```

Comment:
I'm not sure what this does... could you express it in a unit test and/or documentation?